tests/nested/manual/uc-update-assets-secure-add-sbat/task.yaml

```diff
@@ -145,8 +145,19 @@ prepare: |
   remote.wait-for snap-command
   remote.exec "sudo snap wait system seed.loaded"
 
+debug: |
+  cat expected-before
+  cat expected-after
+  for file in $(remote.exec "find /run/mnt/ubuntu-boot/EFI/ -name '*.efi'"); do remote.exec "sha256sum $file"; done
+  for file in $(remote.exec "find /run/mnt/ubuntu-seed/EFI/ -name '*.efi'"); do remote.exec "sha256sum $file"; done
+
 execute: |
-  remote.exec "sha256sum -c" <expected-before
+  if ! result=$(remote.exec "sha256sum -c" <expected-before 2>&1); then
```

Comment:
I looked at the gadget, and 22 ships without bootx64.efi in ubuntu-boot https://github.com/canonical/pc-gadget/blob/9d19aab0860dd2cbd54b0b1e2cb6377a236ede61/gadget/gadget-amd64.yaml#L34-L43 20 does have it https://github.com/canonical/pc-gadget/blob/feb107aa7032a584ac7b8aa60c3a1f02eedecd55/gadget.yaml#L35-L47 hence the difference in expected content which the test does not account for. I think a better fix would be to leave a comment about this fact in the test and check in the context of which UC version the test is being executed

Reply:
Thanks! I will make that change. That sounds much better